syncserver/src/server/test.rs

```diff
@@ -149,7 +149,7 @@ fn create_request(
             let hn = HeaderName::from_lowercase(ln.as_bytes())
                 .expect("Could not get hn in create_request");
             let hv = HeaderValue::from_str(v.as_str()).expect("Could not get hv in create_request");
-            req = req.header(hn, hv);
+            req = req.insert_header((hn, hv));
```

Comment:
The docs claim that `append_header` is equivalent to the old `header` method, but that wasn't true for `TestRequest`.  Here, we need to use `insert_header` to override the header set by `set_json` in order to make the `invalid_content_type` test pass.

Maybe we should keep the calls to `header` and update them in another pass, I'm not sure though.

Reply:
Yeah, the original [`header`](https://docs.rs/actix-web/3.3.3/src/actix_web/test.rs.html#461-469) just set the header in the request directly, and looking at the more recent code [`append_header`](https://docs.rs/actix-web/4.4.1/actix_web/test/struct.TestRequest.html#method.append_header) adds headers of the Key to the request, and [`insert_header`](https://docs.rs/actix-web/4.4.1/actix_web/test/struct.TestRequest.html#method.insert_header) replaces them.

I think we want to use `insert_header` for all of these since we want to create or replace any default header, not just add a value to it.

Reply:
Yeah, that makes sense to me too.  I'll update the code to use `insert_header`